qiskit_dynamics/solvers/solver_classes.py

```diff
@@ -147,8 +235,79 @@ def __init__(
                                present specify as a tuple of lists of frequencies, one for
                                Hamiltonian operators and one for dissipators.
             validate: Whether or not to validate Hamiltonian operators as being Hermitian.
+
+        Raises:
+            QiskitError: If arguments concerning pulse-schedule interpretation are insufficiently
+            specified.
         """
 
+        # set pulse specific information if specified
+        self._hamiltonian_channels = None
+        self._dissipator_channels = None
+        self._all_channels = None
+        self._channel_carrier_freqs = None
+        self._dt = None
+        self._schedule_converter = None
+
+        if any([dt, channel_carrier_freqs, hamiltonian_channels, dissipator_channels]):
+            all_channels = []
+
+            if hamiltonian_channels is not None:
+                hamiltonian_channels = [chan.lower() for chan in hamiltonian_channels]
+                for chan in hamiltonian_channels:
+                    if chan not in all_channels:
+                        all_channels.append(chan)
+                if hamiltonian_operators is None or len(hamiltonian_operators) != len(
+                    hamiltonian_channels
+                ):
+                    raise QiskitError(
+                        """hamiltonian_channels must have same length as hamiltonian_operators"""
+                    )
```

Comment:
Nitpick but you could check first and then do the for loop.

Reply:
yeah good catch - moved this check to before the loop.